src/test/java/uk/gov/justice/digital/domains/DomainExecutorTest.java

```diff
@@ -0,0 +1,528 @@
+package uk.gov.justice.digital.domains;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.spark.SparkConf;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import uk.gov.justice.digital.client.dynamodb.DynamoDBClient;
+import uk.gov.justice.digital.config.BaseSparkTest;
+import uk.gov.justice.digital.config.ResourceLoader;
+import uk.gov.justice.digital.domains.model.DomainDefinition;
+import uk.gov.justice.digital.domains.model.TableDefinition;
+import uk.gov.justice.digital.domains.model.TableInfo;
+import uk.gov.justice.digital.domains.model.TableTuple;
+import uk.gov.justice.digital.domains.service.DomainService;
+import uk.gov.justice.digital.exceptions.DomainExecutorException;
+import uk.gov.justice.digital.service.DeltaLakeService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DomainExecutorTest extends DomainService {
+
+    @TempDir
+    Path folder;
+
+    protected static SparkSession getConfiguredSparkSession(SparkConf sparkConf) {
+        sparkConf
+                .set("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+                .set("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
+                .set("spark.databricks.delta.schema.autoMerge.enabled", "true")
+                .set("spark.databricks.delta.optimizeWrite.enabled", "true")
+                .set("spark.databricks.delta.autoCompact.enabled", "true")
+                .set("spark.sql.legacy.charVarcharAsString", "true");
+
+        return SparkSession.builder()
+                .config(sparkConf)
+                .master("local[*]")
+                .getOrCreate();
+    }
+
+    @Test
+    public void test_getAllSourcesForTable() throws IOException {
```

Comment:
Minor - tests annotated `@Test` don't need to have their name start with the `test_` prefix.